### PR TITLE
Replace paper-item-icon-color with state-icon-color

### PR DESCRIPTION
**Pull request by silamon**
## Breaking change
If you have themes or custom cards that use `--paper-item-icon-color` css variable, consider replacing it with `--state-icon-color`. For now, `--state-icon-color` is filled in with `--paper-item-icon-color`, but it's likely to be removed when Polymer is sunsetted. 

The variables `paper-item-body-two-line-min-height` and `paper-item-body-three-line-min-height` are removed.

## Proposed change
Change `--paper-item-icon-color` with `--state-icon-color`. Other paper-item css variables are removed. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by silamon** [comment]
Monthly revisit: Waiting for a review.

File: src/resources/ha-style.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ const mainStyles = css`
     --label-badge-grey: #9e9e9e;
 
     /* states icon */
-    --state-icon-color: #44739e;
+    --state-icon-color: var(--paper-item-icon-color, #44739e);
```
Comment (bramkragten) [inline src/resources/ha-style.ts:60]: This will still be breaking, as it only applied from the top theme and not elements that are individual themed.

Comment (bramkragten) [reply src/resources/ha-style.ts:60]: I don't necessarily think that is a problem though

File: gallery/src/pages/components/ha-selector.ts [hunk 2]
```diff
@@ -542,9 +542,6 @@ class DemoHaSelector extends LitElement implements ProvideHassElement {
   }
 
   static styles = css`
-    ha-settings-row {
-      --paper-item-body-two-line-min-height: 0;
```
Comment (bramkragten) [inline gallery/src/pages/components/ha-selector.ts:546]: Won't this impact the functionality?

File: src/resources/ha-style.ts (hunk 1 above)
Comment (silamon) [reply src/resources/ha-style.ts:60]: I've tried already many things, but is there a possibility to solve this without breaking change? 
If not, we will need to collect these changes to when Polymer is removed and do them at once.

Comment (bramkragten) [reply src/resources/ha-style.ts:60]: If we change all the uses of `state-icon-color` to have a default, like:
`var(--state-icon-color, #44739e);`, remove it from `ha-style` and put `state-icon-color`: `paper-item-icon-color` in derived styles, I _THINK_ it should not break anything, but not 100% sure...

File: gallery/src/pages/components/ha-selector.ts (hunk 2 above)
Comment (silamon) [reply gallery/src/pages/components/ha-selector.ts:546]: This does make the ha-settings-row in height longer.

**Comment by silamon** [comment]
Since there are way more css variables that will end up, I will reopen the PR closer to when Polymer is ready to be removed. Now there's still the sidebar and the tabs.

**Pull request closed without merging**